pkgs/development/compilers/ocaml/generic.nix

```diff
@@ -0,0 +1,81 @@
+{ version, url, sha256, ...}@args:
+let
+  safeX11 = stdenv: !(stdenv.isArm || stdenv.isMips);
+in
+
+{ stdenv, fetchurl, ncurses, buildEnv, libX11, xproto, useX11 ? safeX11 stdenv }:
+
+assert useX11 -> !stdenv.isArm && !stdenv.isMips;
+
+let
+   useNativeCompilers = !stdenv.isMips;
+   inherit (stdenv.lib) optionals optionalString;
+   name = "ocaml-${version}";
+in
+
+stdenv.mkDerivation (args // rec {
+
+  x11env = buildEnv { name = "x11env"; paths = [libX11 xproto]; };
+  x11lib = x11env + "/lib";
+  x11inc = x11env + "/include";
+
+  inherit name;
+
+  src = fetchurl {
```

Comment:
Ok.

I just splitted the version into three arguments (major, minor and patch) as it needs to be splitten to build the download url.